src/app/triggerFromForm.ts

```diff
@@ -0,0 +1,120 @@
+import {Tickets} from 'node-zendesk';
+
+import {AppContext, AppCallValues} from 'mattermost-redux/types/apps';
+
+import {SubscriptionFields, TriggerFields} from '../utils/constants';
+import {FieldValidationErrors} from '../utils/call_responses';
+
+interface ITriggerFromFrom {
+    getTrigger(): Tickets.CreatePayload;
+    fieldValidationErrors: FieldValidationErrors;
+}
+
+export class TriggerFromForm implements ITriggerFromFrom {
+    values: AppCallValues;
+    context: AppContext;
+    trigger: any
+    fieldValidationErrors: FieldValidationErrors
+
+    constructor(values: AppCallValues, context: AppContext) {
+        this.values = values;
+        this.context = context;
+        this.fieldValidationErrors = {};
+        this.trigger = {};
+    }
+
+    getTrigger(): any {
+        // If not a new trigger, add the trigger ID to the payload
+        // This is a signal to update the trigger, not create a new one
+        if (!this.isNewTrigger()) {
+            const subID = this.values[SubscriptionFields.SubSelect_Name].value;
+            this.addTriggerField('id', subID);
+        }
+        this.addTriggerField('title', this.getTriggerTitle());
+        this.addTriggerField('description', this.getTriggerDescription());
+        this.addTriggerField('actions', this.getTriggerActions());
+        this.addTriggerField('conditions', this.getTriggerConditions());
+
+        return {trigger: this.trigger};
+    }
+
+    getTriggerActions(): any[] {
+        const actions = [
+            {
+                field: TriggerFields.ActionField,
+                value: [
+                    TriggerFields.TargetID,
+                    this.getJSONDataFields(),
+                ],
+            },
+        ];
+        return actions;
+    }
+
+    // getJSONDataFields constructs the object text string for a trigger
+    getJSONDataFields(): string {
```

Comment:
nit: add `trigger` to the function name

Reply:
I debated on the function names in this file before. Filename, interface, factory function, and class name all reference `Trigger` in their names.  I think every non-exported function could actually go the other way and be minimalized by removing all references to a trigger.

For example changing all of the following: 
getTriggerActions -> getActions()
getTriggerTitle -> getTitle()
getTriggerDescription -> getDescription()
getTriggerConditions -> getConditions()

Thoughts? Seems personal preferences differ in this regard, but I do believe consistency is important

Reply:
The reason I mentioned `getJSONDataFields` is with ine other exception all the other methods have Trigger in their name. I guess to maintain consistency we could use this prefix in all of them or in none (excepting the constructor?)

Reply:
done. I removed trigger from the names, minus the exceptions you mentioned.